### PR TITLE
enhance: Add meta pkg for datacoord

### PR DESCRIPTION
**Pull request by congqixia**
See also #28861

Add meta package to replace old one in next PRs.

### CODE REVIEW

File: internal/datacoord/meta/channel/manager.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+func (m *channelManager) DropChannelCheckpoint(ctx context.Context, vChannel string) error {
+	m.mut.Lock()
+	defer m.mut.Unlock()
+	err := m.catalog.DropChannelCheckpoint(ctx, vChannel)
+	if err != nil {
+		return err
+	}
+	delete(m.checkpoints, vChannel)
+	log.Info("DropChannelCheckpoint done", zap.String("vChannel", vChannel))
```
Comment (XuanYang-cn) [inline internal/datacoord/meta/channel/manager.go:93]: please remove cp metrics after drop

File: internal/datacoord/meta/compaction/compaction_task_meta.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	"github.com/milvus-io/milvus/internal/proto/datapb"
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/milvus-io/milvus/pkg/util/timerecord"
+)
+
+type compactionTaskMeta struct {
+	sync.RWMutex
+	ctx     context.Context
+	catalog metastore.DataCoordCatalog
+	// currently only clustering compaction task is stored in persist meta
```
Comment (XuanYang-cn) [inline internal/datacoord/meta/compaction/compaction_task_meta.go:36]: All tasks are stored, not only clustering compaction

File: internal/datacoord/meta/segment/segment_info.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+		for _, deltaLogs := range s.GetDeltalogs() {
+			for _, l := range deltaLogs.GetBinlogs() {
+				rc += l.GetEntriesNum()
+			}
+		}
+		s.deltaRowcount.Store(rc)
+	}
+	r := s.deltaRowcount.Load()
+	return r
+}
```
Comment (XuanYang-cn) [inline internal/datacoord/meta/segment/segment_info.go:189]: deltaRowCount might be changed by L0 Compaction now, this cached value might be inaccurate.

File: internal/datacoord/meta/channel/manager.go (hunk 1 above)
Comment (congqixia) [reply internal/datacoord/meta/channel/manager.go:94]: done

File: internal/datacoord/meta/compaction/compaction_task_meta.go (hunk 2 above)
Comment (congqixia) [reply internal/datacoord/meta/compaction/compaction_task_meta.go]: removed

File: internal/datacoord/meta/segment/segment_info.go (hunk 3 above)
Comment (congqixia) [reply internal/datacoord/meta/segment/segment_info.go:194]: Now `Clone` shall invalidate cache

**Pull request closed without merging**